### ISSUE TITLE
Compiler fails on resolving __dir__ inside a static variable?

### ISSUE THREAD

**Issue by NinoFloris** [peachpiecompiler/peachpie#125]
So this file is part of the generated set of code `composer install` produces, it's the only file preventing compilation and an (out of the box) working composer autoloader.

The function isn't the problem the variables are.
Decomposing it even further the problem seems to be in resolving `__DIR__`, removing that also allows compilation to proceed. Is this a bug or a genuine limitation?

```php
<?php

// autoload_static.php @generated by Composer

namespace Composer\Autoload;

class ComposerStaticInitace36e86701b84f0ecc3b67b1dcf079d
{
    public static $files = array (
        '578a43b5a178d809a159cf0ddb335d08' => __DIR__ . '/../..' . '/index.php',
    );

    public static $fallbackDirsPsr4 = array (
        0 => __DIR__ . '/../..' . '/src',
    );

    public static function getInitializer(ClassLoader $loader)
    {
        return \Closure::bind(function () use ($loader) {
            $loader->fallbackDirsPsr4 = ComposerStaticInitace36e86701b84f0ecc3b67b1dcf079d::$fallbackDirsPsr4;

        }, null, ClassLoader::class);
    }
}
```

**Comment by jakubmisek** [comment]
@NinoFloris Thanks for the test case. `__DIR__` inside a static field initializer is really the issue. Will be fixed.

**Comment by jakubmisek** [comment]
are you running latest version? The test case compiles and runs fine for me

**Comment by jakubmisek** [comment]
resolved in `0.8.0`
